pkg/asset/installconfig/aws/validation.go

```diff
@@ -313,10 +323,24 @@ func validateServiceEndpoints(fldPath *field.Path, region string, services []aws
 	if err := utilerrors.NewAggregate(errs); err != nil {
 		allErrs = append(allErrs, field.Invalid(fldPath, services, err.Error()))
 	}
-
 	return allErrs
 }
 
+// validateRegion checks if the given region is supported by AWS.
+func validateRegion(region string) error {
+	ses, err := GetSession()
+	if err != nil {
+		return err
+	}
+	ec2Session := ec2.New(ses)
+	input := &ec2.DescribeRegionsInput{RegionNames: aws.StringSlice([]string{region})}
+	_, err = ec2Session.DescribeRegions(input)
```

Comment:
We still need something here to limit the amount of time that it takes for the request to time out, right? Otherwise we will just wait here for 60 minutes.

Reply:
Not sure. If I pass an invalid region I get error message real quick but there might be a possibility that the API is unreachable.

Reply:
Ah, I see. The `GetSession` function is using the default region of us-east-1. So this won't work for other partitions.